### PR TITLE
Split forking tests

### PR DESCRIPTION
**Pull request by alcuadrado**
This PR disables the Hardhat Network's forking tests in all of the CI jobs, and creates a new one that runs them with a single configuration (i.e. node version and os).

This isolates the CI's instability in a single job, so that any problem with Alchemy doesn't cancel the entire CI run. Furthermore, somehow making less requests in parallel makes the test run more reliable.  We should investigate why this happens, but in the meantime let's go for this approach so that we have a reliable CI once again.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
             packages/*/node_modules
             packages/hardhat-core/test/internal/hardhat-network/stack-traces/compilers
             packages/hardhat-core/test/internal/hardhat-network/provider/.hardhat_node_test_cache
-          key: ${{ runner.os }}-node-${{ matrix.node }}-${{ hashFiles('yarn.lock') }}-${{ hashFiles('packages/hardhat-core/test/internal/hardhat-network/provider/node.ts') }}
+          key: ${{ runner.os }}-node-10-${{ hashFiles('yarn.lock') }}-${{ hashFiles('packages/hardhat-core/test/internal/hardhat-network/provider/node.ts') }}
```
Comment (alcuadrado) [inline .github/workflows/CI.yml:60]: Fixing an error here. This isn't run with a matrix.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -80,9 +80,6 @@ jobs:
       - name: Install
         run: yarn --frozen-lockfile
       - name: Run tests
-        env:
```
Comment (alcuadrado) [inline .github/workflows/CI.yml:83]: Deleting these env variables disables the forking tests.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -147,7 +141,48 @@ jobs:
...
+      - name: Install
+        run: yarn --frozen-lockfile
+      - name: Clean
+        run: yarn clean
+      - name: Run core tests
         env:
           INFURA_URL: ${{ secrets.INFURA_URL }}
           ALCHEMY_URL: ${{ secrets.ALCHEMY_URL }}
-        run: yarn test
+          DO_NOT_SET_THIS_ENV_VAR____IS_HARDHAT_CI: true
```
Comment (alcuadrado) [inline .github/workflows/CI.yml:183]: This disables the HTTP timeouts

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -147,7 +141,48 @@ jobs:
...
+          FORCE_COLOR: 3
```
Comment (alcuadrado) [inline .github/workflows/CI.yml:184]: We need to force chalk to output colors for Hardhat Network's logging tests to work.

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -147,7 +141,48 @@ jobs:
...
+        run: |
+          cd packages/hardhat-core
+          yarn build
+          npx mocha --recursive "test/internal/hardhat-network/{helpers,jsonrpc,provider}/*.ts"
```
Comment (alcuadrado) [inline .github/workflows/CI.yml:188]: The reason to run them this way is that we only want to run the forking-related tests.

File: packages/e2e/.mocharc.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
   "require": "ts-node/register/files",
   "ignore": ["test/fixture-projects/**/*"],
-  "timeout": 60000
+  "timeout": 180000
```
Comment (alcuadrado) [inline packages/e2e/.mocharc.json:4]: I increased this timeout as it fails very often when running under macos in the ci

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -147,7 +141,48 @@ jobs:
       - name: Clean
         run: yarn clean
       - name: Run tests
+        run: yarn test
+
+  # We should run this tests with multiple configurations
+  # but somehow the requests to Alchemy take much longer when doing that.
+  # As a temporal workaround, we run them with a single config.
```
Comment (feuGeneA) [inline .github/workflows/CI.yml:148]:
```suggestion
  # As a temporary workaround, we run them with a single config.
```

File: .github/workflows/CI.yml [hunk 8]
```diff
@@ -147,7 +141,48 @@ jobs:
...
+  # 
+  # Once we properly refactor Hardhat Network's tests we should
+  # add more configurations (different OS and node versions).
+  #
+  # We should also understand what's going on before blindly refactoring them.
+  test_fork:
+    name: Test Hardhat Network's fork
```
Comment (feuGeneA) [inline .github/workflows/CI.yml:155]:
```suggestion
    name: Test Hardhat Network's forking functionality
```

**Comment by alcuadrado** [comment]
Thanks for the suggestions, @feuGeneA :)

**Pull request merged**